pkg/controller/configmap/config.go

```diff
@@ -316,6 +293,78 @@ func validateConfig(cfg *ini.File, sec string, key string) error {
 	return nil
 }
 
+// Validate node agent specific options
+func validateNodeAgentOptions(configmap *corev1.ConfigMap) []error {
+	errs := []error{}
+	data := &configmap.Data
+	cfg, err := ini.Load([]byte((*data)[operatortypes.ConfigMapDataKey]))
+
+	// Check MTU value
+	mtu := cfg.Section("nsx_node_agent").Key("mtu").Value()
+	_, err = strconv.Atoi(mtu)
+	if err != nil {
+		appendErrorIfNotNil(&errs, errors.Wrapf(err, "mtu is invalid"))
+	}
+	return errs
+}
+
+// Validate common options for both ncp and node agent
+func validateCommonOptions(configmap *corev1.ConfigMap) []error {
+	errs := []error{}
+	data := &configmap.Data
+	cfg, _ := ini.Load([]byte((*data)[operatortypes.ConfigMapDataKey]))
+	// Check DEFAULT section log_file option
+	if cfg.Section("DEFAULT").Key("log_file").Value() != "" {
+		appendErrorIfNotNil(&errs, errors.Errorf("DEFAULT section option: log_file is not allowed to change, current value:%s",
+			cfg.Section("DEFAULT").Key("log_file").Value()))
+	}
+	return errs
+}
+
+// Validate non supported options
+// Including WCP and TAS specific options, MP supported options
+func validateUnSupportedOptions(configmap *corev1.ConfigMap) []error {
+	errs := []error{}
+	data := &configmap.Data
+	is_changed := false
+	cfg, err := ini.Load([]byte((*data)[operatortypes.ConfigMapDataKey]))
+
+	if cfg.Section("nsx_v3").Key("policy_nsxapi").Value() == "True" {
```

Comment:
I guess operator only supports policy_nsxapi==True?

Reply:
Agreed, we can remove this check here, since policy_nsxapi should be default as True.